app-text/mupdf: add 1.23.3

rndxelement opened the pull request:
- Added an ebuild for app-text/mupdf-1.23.3
- Adjusted the patches so that they work for the new version
- There are no changes to the way mupdf is built, as can be seen in the [changelog](https://mupdf.com/releases/history.html)
- Tested use flag combinations with pkg-testing-tool
- Tested reverse dependency app-text/zathura-pdf-mupdf

ceamac reviewed (changes requested):
Any particular reason for keeping so many old ebuilds in the tree?  The oldest one even has a [security bug](https://bugs.gentoo.org/803305) open.

rndxelement commented:
Thanks for the suggestions @ceamac !
I adjusted the following:

- Cleaned the patches and added previous notes
- Added missing deps
- Moved `x11-base/xorg-proto` to DEPEND
- Removed `1.19.1`, `1.21.0` and `1.21.1-r1`, because `1.22.0` is stable already and the older versions aren't required by other ebuilds
